Sources/Support/PaywallExtensions.swift

```diff
@@ -16,65 +16,101 @@ import SwiftUI
+                ids: offering.allProductIdentifiers,
+                prefersPromotionalIcon: prefersPromotionalIcon,
+                icon: icon,
+                placeholderIcon: placeholderIcon
+            )
+            .handlePurchases(offering)
     }
 
 }
 
+@available(iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0, *)
+extension StoreView where Icon == EmptyView, PlaceholderIcon == EmptyView {
+
+    /// Creates a view to load a collection of products from the App Store, and merchandise them.
+    /// When the user purchases products through this paywall, the `RevenueCat` SDK will handle
+    /// the result automatically. All you need to do is to dismiss the paywall
+    public static func forOffering(
+        _ offering: Offering,
+        prefersPromotionalIcon: Bool = false
+    ) -> some View {
+        return self
+            .init(
+                ids: offering.allProductIdentifiers,
+                prefersPromotionalIcon: prefersPromotionalIcon
+            )
+            .handlePurchases(offering)
+    }
+}
+
+// MARK: - SubscriptionStoreView
+
 @available(iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0, *)
 extension SubscriptionStoreView {
 
     /// Creates a ``SubscriptionStoreView`` from an ``Offering``
     /// with custom marketing content.
     /// When the user purchases products through this paywall, the `RevenueCat` SDK will handle
     /// the result automatically. All you need to do is to dismiss the paywall
-    public init(
-        offering: Offering,
+    public static func forOffering(
+        _ offering: Offering,
         @ViewBuilder marketingContent: () -> (Content)
-    ) {
-        self.init(
-            productIDs: offering.subscriptionProductIdentifiers,
-            marketingContent: marketingContent
-        )
+    ) -> some View {
+        return self
+            .init(
+                productIDs: offering.subscriptionProductIdentifiers,
+                marketingContent: marketingContent
+            )
+            .handlePurchases(offering)
     }
 
+}
+
+@available(iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0, *)
+extension SubscriptionStoreView where Content == AutomaticSubscriptionStoreMarketingContent {
+
     /// Creates a ``SubscriptionStoreView`` from an ``Offering``
     /// that doesn't take a custom view to use for marketing content.
-    public init(
-        offering: Offering
-    ) where Content == AutomaticSubscriptionStoreMarketingContent {
-        self.init(productIDs: offering.subscriptionProductIdentifiers)
+    public static func forOffering(_ offering: Offering) -> some View {
+        return self
+            .init(productIDs: offering.subscriptionProductIdentifiers)
+            .handlePurchases(offering)
+    }
+
+}
+
+// MARK: - Private
+
+@available(iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0, *)
+private extension View {
+
+    func handlePurchases(_ offering: Offering) -> some View {
+        return self
+            .onInAppPurchaseStart { product in
```

Comment:
Do we need to clear the cached value on purchase cancellation? 

EDIT: Though thinking about it, it's probably not needed... If the user tries to purchase again with a different offering (super uncommon scenario) it would get set again I guess.

Reply:
Are you good with this then?

Reply:
Yup, sorry for the delay. I thought I had approved it 😅